### ISSUE TITLE
Crashes for two different reasons

### ISSUE THREAD

**Issue by MagikGimp** [Bionus/imgbrd-grabber#2432]
**Steps to reproduce**

CRASH A) Use Pixiv as a source. Unticking this source prevents this crash.
CRASH B) Downloading lots of images (5000+). Stopping downloads triggers the crash.

**Context**

* Provide your `main.log` file to see if any obvious error _occurred_: See attachment.
* Providing your `settings.ini` file will greatly help maintainers reproduce your problem if it may be caused by your configuration: Ditto.
* If the crash _occurred_ during a download, make sure to provide said download as an `.igl` file that Grabber can generate by clicking the `Save` button of the `Downloads` tab: Ditto
* Provide the crash dump that was linked by the Crash Reporter tool: The tool did not launch to capture data for any of the crashes!

**System information**

- OS: Windows 10
- Grabber version: 7.5.1

**Additional context**

Occurred has two Rs...
[Crashes - 27-07-21.zip](https://github.com/Bionus/imgbrd-grabber/files/6902119/Crashes.-.27-07-21.zip)

**Comment by yami-no-tusbas** [comment]
Pixiv work fine for me, as using download list over 10k images, but, looking you log I see this : 
```[17:23:49.382][Info] Loading and saving image from https://ldfclvx.xtnpzbvfdvyy.hath.network:1024/h/5a6e509a9e09a440fce53b35374432dac2899004-283770-884-1277-jpg/keystamp=1627576200-a8739dafaf;fileindex=32111946;xres=2400/13_IMG_0013.jpg in S:\temp\tempp\pix\Naik\.jpg```

On multiple occasion your grabber save a file with and invalid filename for windows (file can not start with "." in windows and filename cannot be null too AFAIK.)

Could you try adding something unique in your filename ? you use : %md5%.%ext% but not all source have md5, maybe try something like : `%id% %md5%.%ext%`

I just downloaded pixiv front page and worked fine. But i'm on nightly channel.
my log : 
```
[23:39:44.600][Info] Loading and saving image from https://i.pximg.net/img-original/img/2021/08/24/06/33/23/92220779_p0.png in D:\Avatars\Pixiv\anonymous - 92220779 - safe - .png
[23:39:44.603][Info] Loading image from https://i.pximg.net/img-original/img/2021/08/24/06/33/23/92220779_p2.png 10
[23:39:44.604][Info] Not enough information to directly load the image (from blacklist: 0 / from file url: 0 / from filename tags: 1/1)
```
I also tried with you file name `%md5%.%ext%` and well, it worked, kinda.
```
[23:45:26.664][Info] Loading image from https://i.pximg.net/img-original/img/2021/08/24/06/39/34/92220832_p0.jpg 10
[23:45:26.666][Info] Loading and saving image from https://i.pximg.net/img-original/img/2021/08/24/06/39/34/92220832_p0.jpg in D:\Avatars\Pixiv\.jpg
[23:45:27.445][Info] Loading image from https://i.pximg.net/img-original/img/2021/08/24/06/39/34/92220832_p1.jpg 10
[23:45:27.447][Info] Loading and saving image from https://i.pximg.net/img-original/img/2021/08/24/06/39/34/92220832_p1.jpg in D:\Avatars\Pixiv\.jpg
[23:45:28.053][Info] Loading image from https://i.pximg.net/img-original/img/2021/08/24/06/39/34/92220832_p2.jpg 10
[23:45:28.055][Info] Loading and saving image from https://i.pximg.net/img-original/img/2021/08/24/06/39/34/92220832_p2.jpg in D:\Avatars\Pixiv\.jpg
```
but the images ended in the prixiv folder as %md5%.%ext% not sure why though. 🤔 Because on first try md5 where absent, maybe grabber calculated them on the fly ? @Bionus we need your light here ! 😋 

As always I'm on windows 11 and nightly ad68aae9

**Comment by Bionus** [comment]
A) Is definitely caused by using an older version, that crash was fixed in 7.6.0 ("Don't crash when trying to load from a source without a valid API"). The fix is to update or login to Pixiv since in earlier versions trying to use Pixiv without a valid login crashed the program.

B) I can't reproduce it, maybe something else is going on?

-----

> but the images ended in the prixiv folder as %md5%.%ext% not sure why though. 🤔 Because on first try md5 where absent, maybe grabber calculated them on the fly ? @Bionus we need your light here ! 😋

%md5% is indeed calculated on the fly when not provided by the source (such as is for Pixiv).

The log is not really correct since the MD5 is not available yet. The path will be re-calculated after the image is downloaded in a temporary file (in the correct folder, but with a temporary "guid" name, such as `490b5cea-f628-4cdf-9a68-e872569140d3.tmp`), then said file will be renamed. I guess the logs aren't very clear about that.
